timemachine/fe/mcgregor.py

```diff
@@ -347,37 +368,35 @@ def recursion(
 
     if connected_core:
         # process g1 using atom_map_1_to_2_information
-        g1_mapped_nodes = set()
-        g1_unmapped_nodes = set()
-        g1_unvisited_nodes = set()
+        g1_node_states = [NODE_STATE_DEMAPPED] * g1.n_vertices
         for a1, a2 in enumerate(atom_map_1_to_2):
             if a1 < layer:
                 # visited nodes
                 if a2 == UNMAPPED:
```

Comment:
Would be good to be consistent about the wording, UNMAPPED vs DEMAPPED

Reply:
They mean different things.

UNMAPPED means either {UNVISITED or VISITED UNMAPPED}
DEMAPPED means *only* {VISITED UNMAPPED}

Reply:
Would be good to add a comment, as it wasn't clear that it was distinct